source/common/router/scoped_config_impl.h

```diff
@@ -132,9 +143,27 @@ class ScopeKeyBuilderImpl : public ScopeKeyBuilderBase {
   std::vector<std::unique_ptr<FragmentBuilderBase>> fragment_builders_;
 };
 
+// ScopedRouteConfiguration and corresponding RouteConfigProvider.
+class ScopedRouteInfo {
+public:
+  ScopedRouteInfo(envoy::api::v2::ScopedRouteConfiguration&& config_proto,
+                  std::unique_ptr<RouteConfigProvider>&& route_provider);
+
+  Router::ConfigConstSharedPtr routeConfig() const { return route_provider_->config(); }
+  const ScopeKey& scopeKey() const { return scope_key_; }
+  const envoy::api::v2::ScopedRouteConfiguration& configProto() const { return config_proto_; }
+  const std::string& scopeName() const { return config_proto_.name(); }
+
+private:
+  const envoy::api::v2::ScopedRouteConfiguration config_proto_;
+  ScopeKey scope_key_;
+  std::unique_ptr<RouteConfigProvider> route_provider_;
```

Comment:
Nit: probably worth introduced `RouteConfigProviderPtr`, this is used a few places.